sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -604,7 +604,10 @@ public final int appendArray(int length) {
    */
   public final int appendStruct(boolean isNull) {
     if (isNull) {
-      appendNull();
+      // This is the same as appendNull but without the assertion for struct types
```

Comment:
why is this necessary?

Reply:
Because `appendNull` itself has an assertion that you are not appending a struct.  So any call to `appendStruct` with `isNull` true would have failed.